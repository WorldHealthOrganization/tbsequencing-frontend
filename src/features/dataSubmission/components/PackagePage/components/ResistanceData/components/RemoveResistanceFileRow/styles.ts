import { css } from '@emotion/react';
import appColors from '../../../../../../../../styles/colors';

const VERTICAL_PADDING = 8;
const HORIZONTAL_PADDING = 16;

export const container = css({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  paddingTop: VERTICAL_PADDING,
  paddingBottom: VERTICAL_PADDING,
  paddingLeft: HORIZONTAL_PADDING,
  paddingRight: HORIZONTAL_PADDING,
  borderBottom: `2px solid ${appColors.tints.whoBlue.lighter}`,
});

export const buttonCell = css({
  display: 'flex',
  justifyContent: 'flex-end',
});

export const sxButtonStyles = {
  fill: appColors.secondary.whoBlue,
  fontSize: 14,
};
