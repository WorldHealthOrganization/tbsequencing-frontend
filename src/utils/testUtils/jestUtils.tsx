import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
