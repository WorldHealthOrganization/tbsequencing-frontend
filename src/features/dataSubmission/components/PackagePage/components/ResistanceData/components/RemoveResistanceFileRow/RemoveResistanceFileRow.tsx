/** @jsxImportSource @emotion/react */
import React from 'react';
import { IconButton } from '@mui/material';
import { Close } from '@mui/icons-material';
import * as styles from './styles';
import OverflowMuiTooltip from '../../../../../../../../components/OverflowMuiTooltip';

interface Props {
  fileName: string;
  onRemoveClick: () => void;
  controlsDisabled: boolean;
}

export const RemoveResistanceFileRow = ({ fileName, onRemoveClick, controlsDisabled }: Props) => (
  <div css={styles.container}>
    <OverflowMuiTooltip tooltip={fileName} text={fileName} />

    {!controlsDisabled && (
    <IconButton
      disabled={controlsDisabled}
      data-testid="remove-btn"
      onClick={onRemoveClick}
    >
      <Close sx={styles.sxButtonStyles} />
    </IconButton>
    )}
  </div>
);

export default RemoveResistanceFileRow;
