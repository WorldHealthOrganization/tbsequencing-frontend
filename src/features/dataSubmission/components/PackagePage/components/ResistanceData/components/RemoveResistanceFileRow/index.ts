import RemoveResistanceFileRow from './RemoveResistanceFileRow';

export default RemoveResistanceFileRow;
