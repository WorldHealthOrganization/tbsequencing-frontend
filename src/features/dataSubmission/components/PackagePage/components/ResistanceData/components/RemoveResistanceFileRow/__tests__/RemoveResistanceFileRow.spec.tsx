import React from 'react';
import { getTreeJSONWithRedux, render, screen } from '../../../../../../../../../utils/testUtils/jestUtils';
import RemoveResistanceFileRow from '../RemoveResistanceFileRow';

describe('RemoveResistanceFileRow component test', () => {
  const mockFileName = 'mockFileName';
  const mockOnRemoveClick = jest.fn();

  test('renders RemoveResistanceFileRow component without error', () => {
    render(
      <RemoveResistanceFileRow fileName={mockFileName} onRemoveClick={mockOnRemoveClick} />,
    );
  });

  it('Should render filename inside row', () => {
    render(
      <RemoveResistanceFileRow fileName={mockFileName} onRemoveClick={mockOnRemoveClick} />,
    );

    screen.getByText(mockFileName);
  });

  it('Should call on remove click callack on button click and render remove button', () => {
    render(
      <RemoveResistanceFileRow fileName={mockFileName} onRemoveClick={mockOnRemoveClick} />,
    );

    const removeBtn = screen.getByTestId('remove-btn');
    removeBtn.click();

    expect(mockOnRemoveClick).toBeCalled();
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <RemoveResistanceFileRow fileName={mockFileName} onRemoveClick={mockOnRemoveClick} />,
    );

    expect(tree).toMatchSnapshot();
  });
});
