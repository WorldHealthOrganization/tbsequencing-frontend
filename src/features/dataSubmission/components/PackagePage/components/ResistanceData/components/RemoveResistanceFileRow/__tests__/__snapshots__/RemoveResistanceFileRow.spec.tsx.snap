// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveResistanceFileRow component test Should render correctly 1`] = `
<div
  className="css-1j3y05o-RemoveResistanceFileRow"
>
  <div
    aria-label="mockFileName"
    aria-labelledby={null}
    className=" css-1nv4tgy-OverflowMuiTooltip"
    data-mui-internal-clone-element={true}
    onBlur={[Function]}
    onFocus={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    mockFileName
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    data-testid="remove-btn"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1fllrtj-MuiSvgIcon-root"
      data-testid="CloseIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
